### PR TITLE
Infinite loop on deliver

### LINKED ISSUE #8824: [deliver] Infinite loop when Apple ID have multiple iTunes Connect team
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I have a Jenkins job about uploading ipa to iTunes Connect, made it for my colleagues who don't have macOS, using this shell command:

```
curl -o appstore.ipa "${URL}"
fastlane fastlane-credentials remove --username "${USERNAME}"
fastlane fastlane-credentials add --username "${USERNAME}" --password "${PASSWORD}"
fastlane deliver --skip_screenshots \
	--skip_metadata \
	--username "${USERNAME}" \
	--ipa appstore.ipa

```

Apple ID belongs to one team works fine, today they use the Apple ID which belongs two team, causes fastlane-deliver infinite loop:

```
Multiple iTunes Connect teams found, please enter the number of the team you want to use: 
Note: to automatically choose the team, provide either the iTunes Connect Team ID, or the Team Name in your fastlane/Appfile:
Alternatively you can pass the team name or team ID using the `FASTLANE_ITC_TEAM_ID` or `FASTLANE_ITC_TEAM_NAME` environment variable

  itc_team_id "xxxxxxxxx"

or

  itc_team_name "aaaaaaaaaa"

1) "aaaaaaaaaa" (xxxxxxxxx)
2) "bbbbbbbbb" (yyyyyyyyy)
Multiple iTunes Connect teams found, please enter the number of the team you want to use: 
Note: to automatically choose the team, provide either the iTunes Connect Team ID, or the Team Name in your fastlane/Appfile:
Alternatively you can pass the team name or team ID using the `FASTLANE_ITC_TEAM_ID` or `FASTLANE_ITC_TEAM_NAME` environment variable

  itc_team_id "xxxxxxxxx"

or

  itc_team_name "aaaaaaaaaa"

1) "aaaaaaaaaa" (xxxxxxxxx)
2) "bbbbbbbbb" (yyyyyyyyy)
Multiple iTunes Connect teams found, please enter the number of the team you want to use: 
Note: to automatically choose the team, provide either the iTunes Connect Team ID, or the Team Name in your fastlane/Appfile:
Alternatively you can pass the team name or team ID using the `FASTLANE_ITC_TEAM_ID` or `FASTLANE_ITC_TEAM_NAME` environment variable

  itc_team_id "xxxxxxxxx"

or

  itc_team_name "aaaaaaaaaa"

1) "aaaaaaaaaa" (xxxxxxxxx)
2) "bbbbbbbbb" (yyyyyyyyy)
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.11.6                                     |
| Ruby                        | 2.4.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.10.1 (Apple Git-78)           |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.11.6 (15G1004)                  |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.1_1/lib          |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | zh_CN.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.26.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.8.1        |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.3        |
| minitest                  | 5.10.1       |
| activesupport             | 4.2.8        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.0.1     |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| json                      | 2.0.2        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| io-console                | 0.4.6        |
| openssl                   | 2.0.3        |
</details>

*generated on:* **2017-04-08**
</details>

### PR DESCRIPTION
**Pull request by 0x5e**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
Raise error when request to choose one iTunes Connect team, and running in non-interactive shell.

### Motivation and Context
Fix #8824

### CODE REVIEW

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 1]
```diff
@@ -139,6 +139,8 @@ def select_team
           puts "#{i + 1}) \"#{team['contentProvider']['name']}\" (#{team['contentProvider']['contentProviderId']})"
         end
 
+        raise "Running in non-interactive shell" if UI.interactive? == false
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: I very much like the intent of this PR, however _spaceship_ doesn't have access to `FastlaneCore::UI`, as they're in separate subfolders, at least until now. Definitely something we could change, as we'd also need this for `credentials_manager`, at both points we don't fail the build if it's not interactive.

**Review by KrauseFx — CHANGES_REQUESTED** [review]

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 1 above)
Comment (ohayon) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: a `UserInterface` class has been added to _spaceship_ so you can change this line to be 
```
raise  "Running in non-interactive shell" unless UserInterface.interactive?
```

**Review by ohayon — CHANGES_REQUESTED** [review]

**Comment by ohayon** [comment]
For some reason this PR is failing in CI but not locally. I am going to push something else up just to force another build because hitting rebuild on Circle is not seeming to work.

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 2]
```diff
@@ -139,6 +139,8 @@ def select_team
           puts "#{i + 1}) \"#{team['contentProvider']['name']}\" (#{team['contentProvider']['contentProviderId']})"
         end
 
+        raise "Running in non-interactive shell" unless UI.interactive?
```
Comment (lacostej) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: We have the same problem under spaceship/lib/spaceship/portal/ui/select_team.rb

**Review by lacostej — CHANGES_REQUESTED** [review]
We should fix it in both places

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 2 above)
Comment (lacostej) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: and there the `interactive?` is redefined

Comment (ohayon) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Good call @lacostej! I didnt realize we had most of this in two places.

**Comment by ohayon** [comment]
Hey @0x5e - wanted to bump this PR to see if you had a chance to look at the comments on here. Let us know if there is anything we can help with!

**Comment by 0x5e** [comment]
@ohayon  I left my company several month ago, now I don't have code & enviroment to test with. Really sorry about this.😔

**Comment by KrauseFx** [comment]
Migrated over to https://github.com/fastlane/fastlane/pull/9981 👍 Thanks again @0x5e for your contribution ❤

**Pull request closed without merging**